airgun/entities/subscription.py

```diff
@@ -0,0 +1,180 @@
+        try:
+            view.browser.wait_for_element(locator, visible=True)
+        except NoSuchElementException:
+            return []
+        return [elem.text for elem in view.browser.elements(locator)]
+
+    def update(self, entity_name, values):
+        # This operation was never implemented in Robottelo (no test
+        # requires it). It is here for consistency, but raises
+        # exception until it is actually needed
+        raise NotImplementedError("Subscriptions update is not implemented")
+
+    def delete(self, entity_name):
+        view = self.navigate_to(self, 'All')
+        for row in view.table.rows(name=entity_name):
+            row['Select all rows'].fill(True)
+        view.delete_button.click()
+        view.confirm_deletion.confirm()
+        self._wait_for_process_to_finish(has_manifest=True)
+
+
+@navigator.register(SubscriptionEntity, 'All')
+class SubscriptionList(NavigateStep):
+    VIEW = SubscriptionListView
+
+    def step(self, *args, **kwargs):
+        self.view.menu.select('Content', 'Subscriptions')
+
+
+@navigator.register(SubscriptionEntity, 'Manage Manifest')
+class ManageManifest(NavigateStep):
+    VIEW = ManageManifestView
+
+    prerequisite = NavigateToSibling('All')
+
+    def step(self, *args, **kwargs):
+        self.parent.manage_manifest_button.click()
+
+    def post_navigate(self, _tries, *args, **kwargs):
+        return self.view.is_displayed
+
+
+@navigator.register(SubscriptionEntity, 'Delete Manifest Confirmation')
+class DeleteManifestConfirmation(NavigateStep):
+    VIEW = DeleteManifestConfirmationView
+
+    prerequisite = NavigateToSibling('Manage Manifest')
+
+    def step(self, *args, **kwargs):
+        wait_for(
+                lambda: self.view.browser.get_attribute(
+                        "disabled",
+                        self.parent.manifest.delete_button) is None,
+                logger=self.view.logger,
+                timeout=10
+        )
+        self.parent.manifest.delete_button.click()
+
+    def post_navigate(self, _tries, *args, **kwargs):
+        return self.view.is_displayed
+
+
+@navigator.register(SubscriptionEntity, 'Add')
+class AddSubscription(NavigateStep):
+    VIEW = AddSubscriptionView
+
+    prerequisite = NavigateToSibling('All')
+
+    def am_i_here(self, *args, **kwargs):
+        return self.view.browser.selenium.current_url.endswith('/add')
+
+    def step(self, *args, **kwargs):
+        self.parent.add_button.click()
+
+
+@navigator.register(SubscriptionEntity, 'Details')
+class SubscriptionDetails(NavigateStep):
+    VIEW = SubscriptionDetailsView
+
+    def prerequisite(self, *args, **kwargs):
+        return self.navigate_to(self.obj, 'All')
+
+    def am_i_here(self, *args, **kwargs):
+        subscription_name = kwargs.get('subscription')
+        if not subscription_name:
+            raise ValueError('Invalid value of subscription parameter')
+        return (self.view.is_displayed and
```

Comment:
does it suffice your needs https://github.com/SatelliteQE/airgun/blob/master/airgun/navigation.py#L73?

Reply:
I tried to use it, but unfortunately it does not cut in. In relevant Robottelo test ([source](https://github.com/SatelliteQE/robottelo/pull/6265/files#diff-9bef302b55c1860e808fb7b1a9bf8094R136)) I first create product. The way Satellite display breadcrumbs there will fool this method into thinking that detailed product page is Subscription detail page (because both happen to have the same string as second breadcrumb item).